auslib/web/common/history_all.py

```diff
@@ -0,0 +1,134 @@
+    try:
+        return history_helper.get_unlimted_histories()
+    except (ValueError, AssertionError) as msg:
+        log.warning("Bad input: %s", msg)
+        return problem(400, "Bad Request", "Error occurred when trying to fetch histories",
+                       ext={"exception": str(msg)})
+
+
+def get_rules_histories():
+    history_table = dbo.rules.history
+    return _get_histories(history_table, get_rules)
+
+
+def get_releases_histories():
+    history_table = dbo.releases.history
+    return _get_histories(history_table, get_releases, process_release_revisions)
+
+
+def get_permissions_histories():
+    history_table = dbo.permissions.history
+    get_permissions = UsersView().get()
+    return _get_histories(history_table, get_permissions)
+
+
+def get_permissions_scheduled_change_histories():
+    """GET /history/scheduled_changes/permissions"""
+    return PermissionScheduledChangeHistoryView().get_all()
+
+
+def get_rules_scheduled_change_histories():
+    """GET /history/scheduled_changes/permissions"""
+    return RuleScheduledChangeHistoryView().get_all()
+
+
+def get_releases_scheduled_change_histories():
+    """GET /history/scheduled_changes/permissions"""
+    return ReleaseScheduledChangeHistoryView().get_all()
+
+
+def get_product_required_signoffs_histories():
+    return ProductRequiredSignoffsHistoryAPIView().get_all()
+
+
+def get_permissions_required_signoffs_histories():
+    return PermissionsRequiredSignoffsHistoryAPIView().get_all()
+
+
+def _get_input_dict():
+    args = request.args
+    obj_keys = []
+    query_keys = []
+    query = {}
+    for key in args:
+        if args.get(key) == 'TRUE':
+            obj_keys.append(key)
+        else:
+            if key != 'limit' and key != 'page':
+                query_keys.append(key)
+
+    for key in query_keys:
+        query[key] = request.args.get(key)
+    return jsonify(query_keys=query_keys, obj_keys=obj_keys, query=query)
+
+
+def filter_helper(obj):
+    req = _get_input_dict()
+    keys = req['keys']
+    keys_values = req['values']
+    for key in keys:
+        if obj[key] != keys_values[key]:
+            return False
+    return True
+
+
+def method_constants():
+    return {
+        'rules': get_rules_histories(),
+        'releases': get_releases_histories(),
+        'permissions': get_permissions_histories(),
+        'permissions_required_signoffs_histories': get_permissions_required_signoffs_histories(),
+        'product_required_signoffs_histories': get_product_required_signoffs_histories(),
+        'releases_scheduled_change_histories': get_releases_scheduled_change_histories(),
+        'rules_scheduled_change_histories': get_rules_scheduled_change_histories(),
+        'permissions_scheduled_change_histories': get_permissions_scheduled_change_histories(),
+    }
+
+
+def get_filtered_history():
+    requested_histories = json.loads(_get_input_dict().data)['obj_keys']
```

Comment:
The way this is implemented now, it looks like each different object needs to be explicitly specified to have its history returned. This is going to be a bit of a maintenance burden (whenever we add new objects, we'll need to update the UI to change the query it makes). What do you think about retrieving history for all objects _except_ if it's explicitly disabled. For example, passing "rules_history=0" would cause rules history not be returned, but not passing "rules_history" at all would return it.